tests/results/test_result_reader.py

```diff
@@ -188,6 +189,11 @@ def test_global_sensitivity_analysis_results_reader(
         qoi_data_index=0,
     )
 
+    # Test equality check.
+    assert results == results
```

Comment:
Just to make sure they are different instances:

```suggestion
    assert results == attr.evolve(results)
```

Reply:
To elaborate, if you don't implement `__eq__` at all it will compare by identity, and the tests as written previously would pass:

```python
>>> class C:
...     def __init__(self, x): self.x=x
... 
>>> c = C(10)
>>> c == c
True
>>> c != object()
True
>>> c != C(20)
True
>>> c == C(10)
False
```

Note that I did not implement `__eq__` at all -- comparing with a copy ensures that `==` is actually correct (the last check is against a copy which is equal but does not compare equal because `__eq__` was not implemented).

TLDR always test `__eq__` against copies.

Reply:
> ...  if you don't implement __eq__ at all it will compare by identity...

@nicoddemus forgot to mention that it is this way be cause `object` implement `__eq__` by identity.
`__eq__` can be overwritten by any class an will be inherited.